imagetagger/imagetagger/images/views.py

```diff
@@ -692,6 +694,37 @@ def dl_script(request):
                             }, content_type='text/plain')
 
 
+def download_imageset_zip(request, image_set_id):
+    image_set = get_object_or_404(ImageSet, id=image_set_id)
+
+    if not settings.ALLOW_ZIP_DOWNLOAD:
+        return HttpResponse(status=HTTP_404_NOT_FOUND)
+
+    if not image_set.has_perm('read', request.user):
+        return HttpResponseForbidden()
+
+    if image_set.image_count == 0:
+        # It should not be possible to download empty image sets. This
+        # is already blocked in the UI, but it should also be checked
+        # on the server side.
+        return HttpResponse(status=HTTP_204_NO_CONTENT)
+
+    if image_set.zip_state != ImageSet.ZipState.READY:
+        return HttpResponse(content=b'Imageset is currently processed', status=HTTP_202_ACCEPTED)
+
+    if settings.USE_NGINX_IMAGE_PROVISION:
```

Comment:
I am worrying about this: While we definitely want to do this, we may need to think about active downloads when a file gets dirty. We may want to unlink the old file and check whether nginx uses the old file descriptor to complete the original download; we might need to consider versioning within the ZIP file names.

Reply:
Zip files are now built in a temporary location before moving them to the final location to avoid download of incomplete files